crates/ruff/src/rules/pep8_naming/rules/invalid_module_name.rs

```diff
@@ -49,18 +54,20 @@ pub fn invalid_module_name(path: &Path, package: Option<&Path>) -> Option<Diagno
     }
 
     if let Some(package) = package {
-        let module_name = if path.file_name().map_or(false, |file_name| {
-            file_name == "__init__.py"
-                || file_name == "__init__.pyi"
-                || file_name == "__main__.py"
-                || file_name == "__main__.pyi"
-        }) {
+        let module_name = if is_module_file(path) {
             package.file_name().unwrap().to_string_lossy()
         } else {
             path.file_stem().unwrap().to_string_lossy()
         };
 
         if !is_module_name(&module_name) {
+            // As a special case, we allow files in `versions` and `migrations` directories to start
+            // with a digit (e.g., `0001_initial.py`), to support common conventions used by Django
+            // and other frameworks.
+            if is_migration_file(path) && is_migration_name(&module_name) {
+                return None;
+            }
```

Comment:
@JonathanPlasse - What do you think about this? It lets us keep the "project convention" logic out of `ruff_python_stdlib` (and keeps `is_module_name` simple since it doesn't rely on directory in any way)... but it does mean we have to check the name twice for valid migrations. However, the most common path is that a module has a valid name anyway, so it may not matter in practice.

Reply:
It is great. Good separation of concerns.